robot_smach_states/benchmark/benchmark_grasping.py

```diff
@@ -0,0 +1,184 @@
+from robot_skills.util.entity import Entity
+from robot_skills.util.kdl_conversions import VectorStamped
+from robot_smach_states import Grab, Inspect, ClassificationResult, NavigateToWaypoint, ForceDrive, SetGripper, \
+    Say
+
+try:
+    from typing import List
+except ImportError:
+    pass
+
+def single_item(robot, results_writer, cls, support, waypoint, inspect_from_area=None, non_strict_class=False):
+    grasp_cls = ds.Designator(cls, name='grasp_cls')
+    support_entity = ds.EdEntityDesignator(robot, id=support, name='support_entity')
+    entity_ids = ds.VariableDesignator([], resolve_type=[ClassificationResult], name='entity_ids')
+    waypoint = ds.EdEntityDesignator(robot, id=waypoint, name='waypoint')
+
+    arm = ds.LockingDesignator(ds.UnoccupiedArmDesignator(robot, {}))
+    arm.lock()
+
+    record = {'robot': robot.robot_name, 'start_waypoint': waypoint, 'expected_class': cls,
+              'id': None}
+
+    try:
+        nav_to_start = NavigateToWaypoint(robot,
+                                          waypoint_designator=waypoint,
+                                          look_at_designator=support_entity)
+
+        assert nav_to_start.execute() == 'arrived', "I did not arrive at the start"
+
+        inspect = Inspect(robot=robot, entityDes=support_entity, objectIDsDes=entity_ids,
+                          navigation_area=inspect_from_area)
+
+        record['inspect_start'] = time.time()
+        assert inspect.execute() == 'done', "I could not inspect the support entity"
+        record['inspect_end'] = time.time()
+        record['inspect_duration'] = '{:3.3f}'.format(record['inspect_end'] - record['inspect_start'])
+
+        inspection_result = entity_ids.resolve()  # type: List[ClassificationResult]
+        if inspection_result:
+            if grasp_cls.resolve() == '*' or non_strict_class:
+                rospy.loginfo("Any item will match")
+                matching_results = inspection_result
+            else:
+                matching_results = [result for result in inspection_result if result.type == grasp_cls.resolve()]
+                rospy.loginfo("Found {} items of class {}".format(len(matching_results), grasp_cls.resolve()))
+
+            if matching_results:
+                if len(matching_results) > 1:
+                    rospy.logwarn("There are multiple items OK to grab, will select the last one")
+                    record['observed_class'] = ' '.join([result.type for result in matching_results])
+                else:
+                    record['observed_class'] = matching_results[-1].type
+                selected_entity_id = matching_results[-1].id
+
+                rospy.loginfo("Selected entity {} for grasping".format(selected_entity_id))
+                grasp_entity = ds.EdEntityDesignator(robot, id=selected_entity_id)
+                record['id'] = selected_entity_id[:6]
+
+                entity = grasp_entity.resolve()  # type: Entity
```

Comment:
This could be done without a designator right?

Reply:
It could with `robot.ed.get_entity(id=...)` but then I'd be replicating what happens in this designator I need anyways